lib/matplotlib/tests/test_text.py

```diff
@@ -962,3 +962,19 @@ def test_text_antialiased_on_default_vs_manual(fig_test, fig_ref):
 
     mpl.rcParams['text.antialiased'] = True
     fig_ref.text(0.5, 0.5, '6 inches x 2 inches')
+
+
+@check_figures_equal()
+def test_text_math_antialiased_on_default_vs_manual(fig_test, fig_ref):
+    fig_test.text(0.5, 0.5, r"OutsideMath $I\'m \sqrt{2}$", antialiased=True)
+
+    mpl.rcParams['text.antialiased'] = True
+    fig_ref.text(0.5, 0.5, r"OutsideMath $I\'m \sqrt{2}$")
+
+
+@check_figures_equal()
+def test_text_math_antialiased_off_default_vs_manual(fig_test, fig_ref):
+    fig_test.text(0.5, 0.5, r"OutsideMath $I\'m \sqrt{2}$", antialiased=False)
+
+    mpl.rcParams['text.antialiased'] = False
+    fig_ref.text(0.5, 0.5, r"OutsideMath $I\'m \sqrt{2}$")
```

Comment:
Unfortunately, neither of these tests fail without this PR, so I'm not sure if they are testing what you want?

Reply:
Yes.
Before this: whether we are using antialiasing is resolved when rendering (`_mathtext.py::Output.to_raster()`). Two test figures are saved after we executing theses codes, resulting two figures resolve to the same `rcParams`.
After this: When creating the text, antialiasing states are set by `True` or `False` if the user specifies the keyword argument. Otherwise, the state resolves to `rcParams` also at the creation time. That is, we no longer query `rcParams` when rendering.